test/groovy/ChecksPublishResultsTest.groovy

```diff
@@ -0,0 +1,181 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertTrue
+
+import util.Rules
+
+class ChecksPublishResultsTest extends BasePipelineTest {
+    Map publisherStepOptions
+    List archiveStepPatterns
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(Rules.getCommonRules(this))
+
+    def stepUnderTest
+
+    @Before
+    void init() {
+        publisherStepOptions = [:]
+        archiveStepPatterns = []
+        // prepare checkResultsPublish step
+        stepUnderTest = loadScript("checksPublishResults.groovy").checksPublishResults
```

Comment:
In another review I suggested the name pattern "examinee". At the moment the name pattern "<stepName>Script" seems to be en vogue (see e.g. ```test/groovy/InfluxWriteDataTest.groovy``)`.